I'm still getting `FAILED h5io/tests/test_io.py::test_state_with_pathlib - AttributeError: property '_raw_path' of 'PosixPath' object has no setter` ?

--- a/pkgs/development/python-modules/h5io/default.nix
+++ b/pkgs/development/python-modules/h5io/default.nix
@@ -2,25 +2,30 @@
   lib,
   buildPythonPackage,
   fetchFromGitHub,
-  pythonOlder,
-  setuptools,
-  numpy,
+
+  # build-system
+  setuptools-scm,
+
+  # dependencies
   h5py,
+  numpy,
+
+  # tests
   pytestCheckHook,
+  scipy,
+  tables,
 }:
 
 buildPythonPackage rec {
   pname = "h5io";
-  version = "0.2.1";
+  version = "0.2.5";
   pyproject = true;
-
-  disabled = pythonOlder "3.7";
 
   src = fetchFromGitHub {
     owner = "h5io";
     repo = "h5io";
     tag = "h5io-${version}";
-    hash = "sha256-3mrHIkfaXq06mMzUwudRO81DWTk0TO/e15IQA5fxxNc=";
+    hash = "sha256-ZkG9e7KtDvoRq9XCExYseE+Z7tMQTWcSiwsSrN5prdI=";
   };
 
   postPatch = ''
@@ -30,22 +35,31 @@ buildPythonPackage rec {
       --replace "--cov=h5io" ""
   '';
 
-  nativeBuildInputs = [ setuptools ];
+  build-system = [ setuptools-scm ];
 
-  propagatedBuildInputs = [
-    numpy
+  dependencies = [
     h5py
+    numpy
   ];
 
-  nativeCheckInputs = [ pytestCheckHook ];
+  nativeCheckInputs = [
+    pytestCheckHook
+    scipy
+    tables
+  ];
+
+  disabledTests = [
+    # See https://github.com/h5io/h5io/issues/86
+    "test_state_with_pathlib"
+  ];
 
   pythonImportsCheck = [ "h5io" ];
 
-  meta = with lib; {
+  meta = {
     description = "Read and write simple Python objects using HDF5";
     homepage = "https://github.com/h5io/h5io";
     changelog = "https://github.com/h5io/h5io/releases/tag/h5io-${version}";
-    license = licenses.bsd3;
-    maintainers = with maintainers; [ mbalatsko ];
+    license = lib.licenses.bsd3;
+    maintainers = with lib.maintainers; [ mbalatsko ];
   };
 }

